### PR TITLE
Modify preprocessing script

### PR DESCRIPTION
**Pull request by sandrinebedard**
# Description

This PR modifies the preprocessing to:

- Add CSA computation of soft and hard GT
- Create disc labels, crop and add them to `data_processed_clean`
- Check if there are NaN values in generated softseg 
- Thresholding softsegs between 0 and 1

An example of results can be found under `duke/projects/contrast-agnostic-seg/contrast-agnostic-test-2022-08-09-all`

### CODE REVIEW

File: process_data.sh [hunk 1]
```diff
@@ -259,107 +285,107 @@ for contrast in "${contrasts[@]}"; do
...
+# Loop through available contrasts
+for file_path in "${inc_contrasts[@]}";do
+  # Find contrast to do segmentation
+  if [[ $file_path == *"T1w"* ]];then
+      contrast_seg="t1"
+  elif [[ $file_path == *"T2star"* ]];then
+      contrast_seg="t2s"
+  elif [[ $file_path == *"T1w_MTS"* ]];then
+      contrast_seg="t1"
+  elif [[ $file_path == *"MTon_MTS"* ]];then
```
Comment (jcohenadad) [inline process_data.sh:314]: you can also add the MToff

Comment (sandrinebedard) [reply process_data.sh:314]: As discussed, the segmentation with sct_deepseg_sc of MToff was to bad and would have required a lot of manual segmentation.

Comment (jcohenadad) [inline process_data.sh:314]: isn't the new file names supposed to be something like `*flip-1_mt-on_MTS`?

File: process_data.sh [hunk 2]
```diff
@@ -377,10 +403,33 @@ for file_path in "${inc_contrasts[@]}";do
...
+  # Find contrast to name csa files
+  if [[ $file_path == *"T1w_MTS"* ]];then
+      contrast_seg="T1w_MTS"
+  elif [[ $file_path == *"T1w"* ]];then
+      contrast_seg="T1w"
+  elif [[ $file_path == *"T2w"* ]];then
+      contrast_seg="T2w"
+  elif [[ $file_path == *"T2star"* ]];then
+      contrast_seg="T2star"
+  elif [[ $file_path == *"MTon_MTS"* ]];then
```
Comment (jcohenadad) [inline process_data.sh:418]: same comment as https://github.com/sct-pipeline/contrast-agnostic-softseg-spinalcord/pull/26/files#r946010671

File: process_data.sh (hunk 1 above)
Comment (sandrinebedard) [reply process_data.sh:314]: It is however not changed yet in spine-generic/data-multi-subject : https://github.com/spine-generic/data-multi-subject/issues/116

Comment (sandrinebedard) [reply process_data.sh:314]: As discussed in the last meeting, we will address this in the data-multi-subject in another time.

**Comment by sandrinebedard** [comment]
I think this is ready to merge, since the naming convention of MTS will not be addressed here (a least yet)

**Pull request merged**